Makefile

```diff
@@ -1,3 +1,24 @@
+# If the first argument is "in_docker"...
+ifeq (in_docker,$(firstword $(MAKECMDGOALS)))
+  # use the rest as arguments for "in_docker"
+  MAKE_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  # Ignore the next args
+  $(eval $(MAKE_ARGS):;@:)
+
+  .ONESHELL:
+  in_docker:
+	docker build -f ./docker/Dockerfile.make -t woodpecker/make:local .
+	docker run -it \
+		-e VERSION="$(VERSION)" \
```

Comment:
nice catch ... you just have to move the block who sets this before the if block:

https://github.com/woodpecker-ci/woodpecker/pull/759/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R25-R40

Reply:
I want the defaults to come from inside the docker environment, in case someone updates the docker image.

Not sure how to update the user though, but I had no issues with it so far.

Reply:
let me play around a bit ... I'll see if i can see some nits/downsides/improvements

Reply:
Any update on this?

Reply:
Yes still on my list 😅 , its awesome how much time you can dedicate, i'll try to catch up 🚀